Mage.Client/src/main/java/mage/client/game/GamePanel.java

```diff
@@ -962,6 +939,19 @@ public synchronized void updateGame() {
             exileWindow.loadCardsAndShow(exile, bigCard, gameId);
         }
 
+        // Remove empty ExileView lists, which prevent removing old windowDialogs after rollbacks.
+        exileViewList.removeIf(HashMap::isEmpty);
```

Comment:
It must be a server side check in GameView creation code (ignore empty exile zones).